gematria/experiments/access_pattern_bm/stl_container.h

```diff
@@ -0,0 +1,42 @@
+#ifndef ACCESS_PATTERN_BM_STL_CONTAINER_H_
+#define ACCESS_PATTERN_BM_STL_CONTAINER_H_
+
+#include <immintrin.h>
+
+#include <random>
+#include <vector>
+
+namespace gematria {
+
+static std::default_random_engine generator;
+static std::uniform_int_distribution<int> distribution(0, 1023);
+
+template <typename Container>
+Container *CreateRandomSTLContainer(std::size_t size) {
+  static_assert(std::is_same<typename Container::value_type, int>::value,
+                "Container must hold `int`s.");
+
+  auto elements = std::vector<int>(size);
+  for (int &element : elements) element = distribution(generator);
+  auto container = new Container(elements.begin(), elements.end());
```

Comment:
Simpler:
```
Container container;
auto inserter = std::inserter(container, container.end());
for (int i = 0; i < size; ++i, ++inserter) {
  *inserter = distribution(generator);
}
```

This should work for pretty much all non-associative STL containers.